include/rocksdb/filter_policy.h

```diff
@@ -51,22 +51,23 @@ class FilterBitsBuilder {
   virtual Slice Finish(std::unique_ptr<const char[]>* buf) = 0;
 
   // Calculate num of keys that can be added and generate a filter
-  // <= the specified number of bytes.
-#if defined(_MSC_VER)
-#pragma warning(push)
-#pragma warning(disable : 4702)  // unreachable code
-#endif
-  virtual int CalculateNumEntry(const uint32_t /*bytes*/) {
-#ifndef ROCKSDB_LITE
-    throw std::runtime_error("CalculateNumEntry not Implemented");
-#else
-    abort();
-#endif
-    return 0;
+  // <= the specified number of bytes, though callers (including RocksDB)
+  // should only rely on that for performance, not correctness.
+  // This default implementation is for old API compatibility.
+  virtual size_t CalculateNumEntries(size_t bytes) {
+    bytes = std::min(bytes, size_t{0xffffffff});
+    return static_cast<size_t>(CalculateNumEntry(static_cast<uint32_t>(bytes)));
```

Comment:
As `CalculateNumEntry` is going to be deprecated, should we not even use it in the default implementation here and just leave it pure virtual?

Reply:
RocksDB is calling the new implementation. For other old implementations of the API to keep working, we need this default implementation.

Reply:
okay, I see, if the user has their own filter with `CalculateNumEntry()` implemented, it would still work.